### PR TITLE
Feat(web-react): Introduce the missing input types of `TextField`: `number`, `search`, `tel`, `url`

### PR DESCRIPTION
**Pull request by adamkudrna**

### CODE REVIEW

File: packages/web-react/src/components/TextField/stories/TextFieldType.tsx [hunk 1]
```diff
@@ -5,9 +5,13 @@ import TextField from '../TextField';
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const Story = (props: unknown) => (
   <div style={{ display: 'flex', flexWrap: 'wrap', gap: '1.5rem 1rem' }}>
-    <TextField id="textfield-text" label="Type text" name="textfield-text" type="text" />
     <TextField id="textfield-email" label="Type email" name="textfield-email" type="email" />
+    <TextField id="textfield-number" label="Type number" name="textfield-number" type="number" />
     <TextField id="textfield-password" label="Type password" name="textfield-password" type="password" />
+    <TextField id="textfield-search" label="Type search" name="textfield-search" type="search" />
```
Comment (crishpeen) [inline packages/web-react/src/components/TextField/stories/TextFieldType.tsx:11]: We might wanna check with designers the number arrows and search clear in the future. Now it works :)

File: packages/web-react/src/components/TextFieldBase/README.md [hunk 2]
```diff
@@ -3,24 +3,23 @@
...
+| `isDisabled`      | boolean                                                       | Whether is field disabled         |
+| `isLabelHidden`   | boolean                                                       | Whether is label hidden           |
+| `isMultiline`     | boolean                                                       | Whether is DOM element `textarea` |
+| `isRequired`      | boolean                                                       | Whether is field required         |
+| `label`           | string                                                        | Label text                        |
+| `message`         | string                                                        | Validation or help message        |
+| `name`            | string                                                        | Input name                        |
+| `placeholder`     | string                                                        | Input placeholder                 |
+| `type`            | `email`, `number`, `password`, `search`, `tel`, `text`, `url` | Input type                        |
```
Comment (crishpeen) [inline packages/web-react/src/components/TextFieldBase/README.md:23]: This is also in the `web-twig` package. Can you fix it there as well? Or is it gonna be another PR?

Comment (crishpeen) [reply packages/web-react/src/components/TextFieldBase/README.md:23]: My bad, just saw it. Sorry.

**Review by crishpeen — APPROVED** [review]

File: packages/web-react/src/components/TextField/stories/TextFieldType.tsx (hunk 1 above)
Comment (adamkudrna) [reply packages/web-react/src/components/TextField/stories/TextFieldType.tsx:11]: I think that may have been the reason why these types were not listed here (TODO: drop a comment next time). But:

1. we suggest using them in the `web` package,
2. we need the `number` type in Jobs 🤷🏻.
